cram/cram_io.c

```diff
@@ -3244,14 +3248,34 @@ static char *load_ref_portion(BGZF *fp, ref_entry *e, int start, int end) {
 
     /* Strip white-space if required. */
     if (len != end-start+1) {
-        int i, j;
+        hts_pos_t i, j;
```

Comment:
If these are changed to `hts_pos_t`, don't you also need to change `start` and `end` to be able to use the extra range?  (Also in the callers... Hmm, it does seem to cascade a bit, but would be necessary to properly handle long refs in CRAMv4).

Reply:
Yes I realised it's only a partial fix.  I did initially just leave them as ints, as before, but figured I may as well make them correct at this point while I'm tinkering with the code, even if they're not elsewhere.

It's all a moot point anyway as the file format cannot cope with anything more than int, until we go to CRAM 4, and I don't see that happening any time soon given the lack of another specification maintainer and the many years it took for CRAM 3.1 to get a second implementation.

Do you want me to reverse this line change, or to spend the time to update the rest of the code data types?

Reply:
I'll go for it as-is, but we should really make a follow-up to sort out the other data types.